### DISCUSSION TITLE
silence "has not been processed because no matching rules were found."?

### DISCUSSION THREAD

**Discussion by juju4** [Azure/PSRule.Rules.Azure#2034] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

silence "has not been processed because no matching rules were found."?
or ignore md, png files and so on.

Example: https://github.com/juju4/Azure-Sentinel/actions/runs/4152879216/jobs/7184135632#step:3:58

**Describe the solution you'd like**

* file exclude as per https://github.com/Azure/PSRule.Rules.Azure/issues/2031
* cli argument to ignore invalid files, silently or at lower severity level

**Describe alternatives you've considered**

Leave as is but polluting pipeline output and logs.

**Comment by BernieWhite** [comment, marked as answer]
@juju4 Thanks for raising an issue. I think we can answer this one as a question so I've move it here so that it might help others in the community.

PSRule will detect all files by default, there may be some changes on how this works in PSRule v3.

However today you have a number of options.

- Exclude uninteresting files that you don't want processed by PSRule. We can exclude based on a git path spec see #2033.
- Ignore the not processed warning by setting [Execution.NotProcessedWarning](https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#executionnotprocessedwarning) option.
- Target the input path where PSRule starts it's scan for files at a different location by setting `inputPath` in CI [GitHub Action](https://github.com/microsoft/ps-rule#inputpath) / [Azure Pipelines Task](https://github.com/microsoft/PSRule-pipelines/blob/main/docs/tasks.md#ps-rule-assert). However avoid setting the `inputType` option as this will not work with PSRule for Azure.

--- 

Does this help?

**Reply by BernieWhite** [reply]
@juju4 Did this answer your question. If so please mark as answer.
